stdlib/contrib/qxip/logql/logql.flux

```diff
@@ -0,0 +1,87 @@
+// Package logql provides functions meant to simplify common logql queries.
+//
+// The primary function in this package is `logql.query_range()`
+//
+// ## Metadata
+// introduced: NEXT
+//
+package logql
+
+
+import "csv"
+import "date"
+import "experimental"
+import "experimental/http/requests"
+
+// query_range queries data from a specified LogQL query within given time bounds,
+// filters data by query, timerange, and optional limit expressions.
+//
+// ## Parameters
+// - url: LogQL/qryn URL.
+// - path: LogQL query_range API path.
+// - limit: Query limit. Default is 100.
+// - query: LogQL query to execute.
+// - start: Earliest time to include in results. Default is `-1h`.
+//
+//   Results include points that match the specified start time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `-1h`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
+// - end: Latest time to include in results. Default is `now()`.
+//
+//   Results exclude points that match the specified stop time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `now()`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
+// - step: Query stepping. Default is 10.
+// - orgid: Optional Organization Id for partitioning.
+//
+// ## Examples
+// ### Query specific fields in a measurement from LogQL/qryn
+// ```no_run
+// import "contrib/qxip/logql"
+//
+// logql.query_range(
+//     url: "http://qryn:3100",
+//     path: "/loki/api/v1/query_range",
+//     start: -1h,
+//     end: now(),
+//     query: "{job=\"dummy-server\"}",
+//     limit: 100,
+// )
+// ```
+//
+// ## Metadata
+// tags: inputs
+//
+query_range = (
+        url="http://127.0.0.1:3100",
+        path="/loki/api/v1/query_range",
+        query="",
+        limit=100,
+        step=10,
+        start=-1h,
+        end=now(),
+        orgid="",
+    ) =>
+    {
+        dstart = date.sub(from: start, d: 0d)
+        dend = date.sub(from: end, d: 0d)
+        response =
+            requests.get(
+                url: url + path,
+                params:
+                    [
+                        "query": [query],
+                        "limit": ["${limit}"],
+                        "start": [string(v: uint(v: dstart))],
+                        "end": [string(v: uint(v: dend))],
```

Comment:
Just a nit here. The Loki API does support RFC3339 timestamps, so you don't necessarily need to convert these to Unix nanosecond timestamps, but you can.

Reply:
See comments in the above discussion. At this time, we must use nanosecond ts for compatibility. 
> dstart = date.sub(from: start, d: 0d)
        dend = date.sub(from: end, d: 0d)